source_test.go

```diff
@@ -481,6 +481,54 @@ func TestSourceDockerImage(t *testing.T) {
 	})
 }
 
+func TestPatchSource_MalformedPatch(t *testing.T) {
+	worker := llb.Scratch()
+	sourceState := llb.Scratch().File(
+		llb.Mkfile("/file.txt", 0644, []byte("Hello World")),
+	)
+	sourceToState := map[string]llb.State{
+		"malformed_patch": llb.Scratch().File(
+			llb.Mkfile("/malformed.patch", 0644, []byte("invalid patch content")),
+		),
+	}
+	patchNames := []PatchSpec{
+		{
+			Source: "malformed_patch",
+		},
+	}
+
+	assert.Assert(t, func() {
```

Comment:
It looks like this is testing that the closure is not nil which doesn't look correct.

I think the easiest way to test these is in the integration tests (under `test/`) where we can get it to actually run the patch and produce an error.